rmf_fleet_adapter/src/rmf_fleet_adapter/agv/internal_FleetUpdateHandle.hpp

```diff
@@ -176,13 +172,14 @@ class FleetUpdateHandle::Implementation
   std::unordered_set<std::size_t> available_charging_waypoints = {};
 
   double current_assignment_cost = 0.0;
-  // Map to store task id with assignments for BidNotice
-  std::unordered_map<std::string, Assignments> bid_notice_assignments = {};
+  
+  // Map to store task id with the profile msg of the received tasks
+  using TaskProfile = rmf_task_msgs::msg::TaskProfile;
+  std::unordered_map<std::string, TaskProfile> received_task_profiles = {};
+  
+  // Tuple which stored the latest received BidNotice
+  std::tuple<rmf_task::ConstRequestPtr, Assignments> latest_bid_notice_assignments;
 
-  std::unordered_map<
```

Comment:
Hmm I think keeping track of these will be essential but we will need some ways to maintain a fixed size of these data structures.

Reply:
I have removed all local data structures (use for task caching) in `FleetUpdateHandle` that will keep appending indefinitely. The logic here is to retrieve those caches that are stored within each task_manager, reduces complexity.